src/nexus.cpp

```diff
@@ -160,31 +152,35 @@ void Nexus::start()
  */
 void Nexus::showLogin()
 {
+    loginScreen = new LoginScreen();
+
+#ifdef Q_OS_MAC
+    connect(loginScreen, &LoginScreen::windowStateChanged, this,
+            &Nexus::onWindowStateChanged);
+#endif
+
     delete widget;
     widget = nullptr;
 
     delete mProfile;
     mProfile = nullptr;
 
     loginScreen->reset();
-    loginScreen->move(QApplication::desktop()->screen()->rect().center() - loginScreen->rect().center());
     loginScreen->show();
-    ((QApplication*)qApp)->setQuitOnLastWindowClosed(true);
+    loginScreen->move(QApplication::desktop()->screen()->rect().center() -
```

Comment:
Moving/resizing a window will fail before the **first** show(), which also initializes the widget position and size. This is a logical consequence of the fact, that move/resize events are **deferred** as long as a widget is invisible. Especially on X11 based systems, there are further factors playing a role. See Qt docs for [QWidget::setGeometry](http://doc.qt.io/qt-5/qwidget.html#geometry-prop) and [Window Geometry](http://doc.qt.io/qt-5/application-windows.html#x11-peculiarities) for further information.